### PR TITLE
Show vertical list form directly

### PR DESCRIPTION
**Pull request by yuki-stripe**
## Summary
See commits messages - main change is to show the form directly in the edge case where there is only one payment method, 

## Motivation
https://jira.corp.stripe.com/browse/MOBILESDK-2067

## Testing
See snapshot test

## Changelog
Not user facing

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentMethodFormViewController.swift [hunk 1]
```diff
@@ -27,7 +29,9 @@ class PaymentMethodFormViewController: UIViewController {
 
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
-        _ = self.form.beginEditing()
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentMethodFormViewController.swift:32]: 👍 nit: maybe add a comment about why we do this

**Review by porter-stripe — APPROVED** [review]

**Comment by admsev** [comment]
@gooroodev would you mind to take a look?

**Comment by gooroodev** [comment]
Here to help, @admsev! Thanks for the tag!

{mistakes: 0, typos: 0, security: 0, performance: 0, code_styling: 2, others: 0}

### Summary of Changes
1. Enabled delayed payment methods in playground settings.
2. Added a snapshot test for `VerticalPaymentSheetViewController`.
3. Modified `VerticalPaymentMethodListViewController` to return the number of row buttons.
4. Updated `PaymentMethodFormViewController` to accept a payment method type.
5. Modified `PaymentSheetVerticalViewController` to display the form directly if only one payment method is available and it collects user input.
6. Added helper function `makeForm` in `PaymentSheetVerticalViewController`.
7. Refactored `VerticalPaymentMethodListView` to return row buttons.

### Issues, Bugs, or Typos
No issues, bugs, or typos were found.

### Code Quality and Style
The code quality is good, but there are a couple of improvements for better code readability and maintainability.

### Proposed Improvements
1. **Improvement in `PaymentSheetVerticalViewController`:** 
   - The `makeForm` method should be marked as `private` since it is a helper method intended for internal use.
   
   ```swift
   // file_path:line_number
   private func makeForm(paymentMethodType: PaymentSheet.PaymentMethodType) -> PaymentMethodElement {
       return PaymentSheetFormFactory(intent: intent, configuration: .paymentSheet(configuration), paymentMethod: paymentMethodType).make()
   }
   ```

2. **Improvement in `VerticalPaymentMethodListViewController`:**
   - The `rowCount` property should be marked as `private` if it is not intended to be accessed outside the class.
   
   ```swift
   // file_path:line_number
   private var rowCount: Int {
       return listView.rowButtons.count
   }
   ```

### General Review
The code is well-structured and follows good coding practices. The changes are clear and concise, and the addition of the snapshot test ensures that the functionality is verified visually. The use of helper methods and the careful handling of view controllers improve the readability and maintainability of the code. Overall, the pull request is solid and ready for merging after the minor improvements.

Yours, [Gooroo.dev](https://github.com/apps/gooroo-dev). React or reply to let me know your opinion!

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentMethodFormViewController.swift (hunk 1 above)
Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentMethodFormViewController.swift:32]: I'll follow up in another PR - I might remove/rework this.
